elephant/spike_train_generation.py

```diff
@@ -1066,33 +1122,44 @@ def _cpp_het_stat(A, t_stop, rate, t_start=0. * pq.ms):
 
     # Computation of Parameters of the two CPPs that will be merged
     # (uncorrelated with heterog. rates + correlated with homog. rates)
-    N = len(rate)  # number of output spike trains
-    A_exp = np.dot(A, range(N + 1))  # expectation of A
-    r_sum = np.sum(rate)  # sum of all output firing rates
-    r_min = np.min(rate)  # minimum of the firing rates
-    r1 = r_sum - N * r_min  # rate of the uncorrelated CPP
-    r2 = r_sum / float(A_exp) - r1  # rate of the correlated CPP
-    r_mother = r1 + r2  # rate of the hidden mother process
+    n_spiketrains = len(rates)  # number of output spike trains
+    # amplitude expectation
+    expected_amplitude = np.dot(amplitudes, range(n_spiketrains + 1))
```

Comment:
range is generator
it's better to use np.arange when it's been a part of math operation